sound/soc/sof/ipc.c

```diff
@@ -18,8 +18,10 @@
 #include "sof-audio.h"
 #include "ops.h"
 
-static void ipc_trace_message(struct snd_sof_dev *sdev, u32 msg_type);
-static void ipc_stream_message(struct snd_sof_dev *sdev, u32 msg_cmd);
+typedef void (*ipc_rx_callback)(struct snd_sof_dev *sdev, void *full_msg);
+
+static void ipc_trace_message(struct snd_sof_dev *sdev, void *full_msg);
+static void ipc_stream_message(struct snd_sof_dev *sdev, void *full_msg);
```

Comment:
I don't get the concept of passing a void pointer without any indication of size to the client. How would they know how much data they can read from the 'full_msg' area?